seive.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+function seiveOriginal(n) {
+  let primes = new Array(n - 1).fill(true);
+  [primes[0], primes[1]] = [false, false];
+  for (let i = 2; i <= n; i++) {
+    for (let j = i * i; j <= n; j += i) {
```

Comment:
As far as I understand it's not necessary to do this loop for complex numbers. Meaning that there is should be a check here for whether should be done or not.